build.gradle

```diff
@@ -24,132 +29,405 @@ if (verifySettingsGradle()) {
+    apply from: 'addon.gradle'
 }
 
-// Generate the InternalTags class with the version number as a field
-tasks.injectTags.configure {
-    outputClassName.set("${project.group}.GTInternalTags")
+
+// Repositories
+
+// Allow unsafe repos but warn
+repositories.configureEach { repo ->
+    if (repo instanceof UrlArtifactRepository) {
+        if (repo.getUrl() != null && repo.getUrl().getScheme() == "http" && !repo.allowInsecureProtocol) {
+            logger.warn("Deprecated: Allowing insecure connections for repo '${repo.name}' - add 'allowInsecureProtocol = true'")
+            repo.allowInsecureProtocol = true
+        }
+    }
+}
+
+// Allow adding custom repositories to the buildscript
+if (getFile('repositories.gradle').exists()) {
+    apply from: 'repositories.gradle'
 }
 
 repositories {
-    maven {
-        // MixinBooter
-        name 'Cleanroom Maven'
-        url 'https://maven.cleanroommc.com'
-    }
-    maven {
-        // CraftTweaker and JEI
-        name 'BlameJared Maven'
-        url 'https://maven.blamejared.com'
-    }
-    maven {
-        // TOP, CTM, GRS, AE2
-        name 'Curse Maven'
-        url 'https://www.cursemaven.com'
-        content {
-            includeGroup 'curse.maven'
+    if (includeWellKnownRepositories.toBoolean()) {
+        maven {
+            name 'Cleanroom Maven'
+            url 'https://maven.cleanroommc.com'
+        }
+        maven {
+            name 'BlameJared Maven'
+            url 'https://maven.blamejared.com'
+        }
+        maven {
+            name 'Curse Maven'
+            url 'https://www.cursemaven.com'
+            content {
+                includeGroup 'curse.maven'
+            }
         }
     }
-    maven {
-        // Mixin
-        name 'Sponge Maven'
-        url 'https://repo.spongepowered.org/maven'
+    if (usesMixins.toBoolean() || forceEnableMixins.toBoolean()) {
+        maven {
+            name 'Sponge Maven'
+            url 'https://repo.spongepowered.org/maven'
+        }
+        // need to add this here even if we did not above
+        if (!includeWellKnownRepositories.toBoolean()) {
+            maven {
+                name 'Cleanroom Maven'
+                url 'https://maven.cleanroommc.com'
+            }
+        }
     }
     mavenLocal() // Must be last for caching to work
 }
 
-apply from: 'dependencies.gradle'
 
-//noinspection ConfigurationAvoidance
-for (File at : sourceSets.getByName('main').resources.files) {
-    if (at.name.toLowerCase().endsWith('_at.cfg')) {
-        tasks.deobfuscateMergedJarToSrg.accessTransformerFiles.from at
-        tasks.srgifyBinpatchedJar.accessTransformerFiles.from at
+// Dependencies
+
+// Configure dependency configurations
+configurations {
+    embed
+    implementation.extendsFrom(embed)
+}
+
+dependencies {
+    if (usesMixins.toBoolean() || forceEnableMixins.toBoolean()) {
+        implementation 'zone.rong:mixinbooter:7.0'
+        api('org.spongepowered:mixin:0.8.3') {
```

Comment:
Buildscript Repo: Do we want to use Mixinbooter 8 now that it is released, since Rong said it should solve the incorrect mixin location issue in RFG?